### PR TITLE
refactor: add flags for common config items (db uri, brokers, topic) #24

### PR DESCRIPTION
**Pull request by xbcsmith**

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -170,6 +171,9 @@ func init() {
 	// create two new flags, one for host and one for port
 	rootCmd.Flags().String("host", "localhost", "host to listen on")
 	rootCmd.Flags().String("port", "8042", "port to listen on")
+	rootCmd.Flags().String("brokers", "localhost:9092", "brokers uri")
```
Comment (caproven) [inline cmd/root.go:174]: Docker compose in docs exposes 19092 (current hardcoded value) so changing to 9092 breaks some examples. 9092 is kafka default so I prefer that.

Docs should be revisited - I'm happy to run through them & resolve any issues if you don't want to address that in current PR.

File: cmd/root.go [hunk 2]
```diff
@@ -170,6 +171,9 @@ func init() {
...
+	rootCmd.Flags().String("topic", "epr.dev.events", "topic to produce events on")
+	rootCmd.Flags().String("db", "postgres://localhost:5432", "database connection string")
 
 	rootCmd.Flags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.generic.yaml)")
```
Comment (caproven) [inline cmd/root.go:178]:
```suggestion
	rootCmd.Flags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.epr.yaml)")
```

File: cmd/root.go (hunk 1 above)
Comment (caproven) [inline cmd/root.go:174]: Please indicate plurality, e.g., "broker uris separated by commas"

File: cmd/root.go [hunk 3]
```diff
@@ -154,9 +155,9 @@ func initConfig() {
 		if err != nil {
 			fmt.Print("unable to find home directory")
 		}
-		// Search config in home directory with name ".generic" (without extension).
+		// Search config in home directory with name ".epr" (without extension).
 		viper.AddConfigPath(home)
```
Comment (caproven) [inline cmd/root.go:159]: Not pertinent to PR - should this use xdg config home instead of user home?

Comment (xbcsmith) [reply cmd/root.go:159]: Yes we should. #31

File: cmd/root.go (hunk 1 above)
Comment (xbcsmith) [reply cmd/root.go]: Done

File: README.md [hunk 4]
```diff
@@ -76,6 +76,21 @@ make megalint
 
 ## Usage
 
+```txt
+Usage:
+  epr-server [flags]
+
+Flags:
+      --brokers string   brokers uri (default "localhost:9092")
+      --config string    config file (default is $HOME/.generic.yaml)
```
Comment (caproven) [inline README.md:85]: Sorry, missed this yesterday.
```suggestion
      --config string    config file (default is $HOME/.epr.yaml)
```

**Review by caproven — CHANGES_REQUESTED** [review]
I missed one last reference of `.generic` as the config file in previous review. Otherwise LGTM

**Review by caproven — DISMISSED** [review]
Actually, I'll approve since the config location will change with #31 anyway. Feel free to submit as-is.

File: cmd/root.go [hunk 5]
```diff
@@ -76,14 +61,30 @@ func preRun(cmd *cobra.Command, _ []string) error {
...
+	topic := viper.GetString("topic")
+	host := viper.GetString("host")
+	port := viper.GetString("port")
+
+	dburl, err := url.Parse(viper.GetString("db"))
+	if err != nil {
+		panic(err)
+	}
+
+	dbhost, dbportstr, _ := net.SplitHostPort(dburl.Host)
```
Comment (ninjamast3r) [inline cmd/root.go:75]: Shouldn't ignore what I assume is an error there.

File: cmd/root.go [hunk 6]
```diff
@@ -76,14 +61,30 @@ func preRun(cmd *cobra.Command, _ []string) error {
...
 	logger.V(1).Info("debug enabled")
+	// TODO probably need some better input validation
+	brokers := strings.Split(viper.GetString("brokers"), ",")
+	topic := viper.GetString("topic")
+	host := viper.GetString("host")
+	port := viper.GetString("port")
+
+	dburl, err := url.Parse(viper.GetString("db"))
+	if err != nil {
+		panic(err)
```
Comment (ninjamast3r) [inline cmd/root.go:72]: Don't panic, just return the error.

File: cmd/root.go [hunk 7]
```diff
@@ -76,14 +61,30 @@ func preRun(cmd *cobra.Command, _ []string) error {
...
+	}
+
+	dbhost, dbportstr, _ := net.SplitHostPort(dburl.Host)
+	dbport, err := strconv.Atoi(dbportstr)
+	if err != nil {
+		panic(err)
```
Comment (ninjamast3r) [inline cmd/root.go:78]: same here.

**Review by ninjamast3r — CHANGES_REQUESTED** [review]

**Review by caproven — APPROVED** [review]

**Pull request merged**